crates/aptos-rest-client/src/lib.rs

```diff
@@ -69,46 +73,65 @@ pub struct Client {
     version_path_base: String,
 }
 
-impl Client {
-    pub fn new_with_timeout(base_url: Url, timeout: Duration) -> Self {
-        Client::new_with_timeout_and_user_agent(base_url, timeout, USER_AGENT)
-    }
-
-    pub fn new_with_timeout_and_user_agent(
-        base_url: Url,
-        timeout: Duration,
-        user_agent: &str,
-    ) -> Self {
-        let inner = ReqwestClient::builder()
-            .timeout(timeout)
-            .user_agent(user_agent)
-            .cookie_store(true)
-            .build()
-            .unwrap();
-
-        // If the user provided no version in the path, use the default. If the
-        // provided version has no trailing slash, add it, otherwise url.join
-        // will ignore the version path base.
-        let version_path_base = match base_url.path() {
-            "/" => DEFAULT_VERSION_PATH_BASE.to_string(),
-            path => {
-                if !path.ends_with('/') {
-                    format!("{}/", path)
-                } else {
-                    path.to_string()
-                }
-            },
-        };
+pub struct ClientBuilder {
+    reqwestBuilder: ReqwestClientBuilder,
+    version_path_base: String,
+    base_url: Url, // Default to mainnet if None
```

Comment:
let's default to devnet

Reply:
Let's not default to anything, re my comment on the builder pattern below.